templates/user/apiProfileForm.tpl

```diff
@@ -23,21 +23,37 @@
 
 	{include file="controllers/notification/inPlaceNotification.tpl" notificationId="apiProfileNotification"}
 
-	{fbvFormSection list=true}
+	{* {fbvFormSection list=true}
 		{fbvElement id=apiKeyEnabled type="checkbox" label="user.apiKeyEnabled" checked=$apiKeyEnabled value=1}
 		{fbvElement id=generateApiKey type="checkbox" label="user.apiKey.generate" value=1}
-	{/fbvFormSection}
-	<p>{translate key="user.apiKey.generateWarning"}</p>
+	{/fbvFormSection} *}
 
-	{fbvFormSection}
+	{fbvFormSection title="user.apiKey"}
 		{if !$apiKey}{assign var=apiKey value="common.none"|translate}{/if}
-		{fbvElement id=apiKey type="text" label="user.apiKey" readonly="true" value=$apiKey size=$fbvStyles.size.MEDIUM}
+		{fbvElement id=apiKey type="text" readonly="true" inline=true value=$apiKey size=$fbvStyles.size.MEDIUM}
+		{if !$apiSecretMissing}
+			{fbvElement id=apiKeyAction type="hidden" readonly="true" value=$apiKeyAction}
+			<button
+				type="submit"
+				{if $apiKeyAction === \PKP\user\form\APIProfileForm::API_KEY_DELETE}
+					onClick="return confirm({translate|json_encode|escape key='user.apiKey.remove.confirmation.message'})"
+					class="pkpButton pkpButton--isWarnable"
+				{else}
+					class="pkp_button pkp_button_primary"
+				{/if}
+			>
+				{translate key=$apiKeyActionTextKey}
+			</button>
+		{/if}
+		<p>
+			{translate key=($apiKeyAction) ? "user.apiKey.generateWarning" : "user.apiKey.removeWarning"}
```

Comment:
If you're using defined constants, I'd suggest using them comprehensively rather than mixing them with truthy `if`s; this should check for the named constants too.

Reply:
fixed.